packages/server/events-processing-platform/subscriptions/notifications/organization_event_handler.go

```diff
@@ -0,0 +1,89 @@
+package notifications
+
+import (
+	"context"
+
+	"github.com/openline-ai/openline-customer-os/packages/server/events-processing-platform/config"
+	"github.com/openline-ai/openline-customer-os/packages/server/events-processing-platform/domain/organization/events"
+	"github.com/openline-ai/openline-customer-os/packages/server/events-processing-platform/eventstore"
+	"github.com/openline-ai/openline-customer-os/packages/server/events-processing-platform/graph_db"
+	"github.com/openline-ai/openline-customer-os/packages/server/events-processing-platform/graph_db/entity"
+	"github.com/openline-ai/openline-customer-os/packages/server/events-processing-platform/logger"
+	"github.com/openline-ai/openline-customer-os/packages/server/events-processing-platform/repository"
+	"github.com/openline-ai/openline-customer-os/packages/server/events-processing-platform/tracing"
+	"github.com/opentracing/opentracing-go"
+	"github.com/pkg/errors"
+)
+
+type OrganizationEventHandler struct {
+	repositories         *repository.Repositories
+	log                  logger.Logger
+	notificationProvider NotificationProvider
+	cfg                  *config.Config
+}
+
+func NewOrganizationEventHandler(log logger.Logger, repositories *repository.Repositories, cfg *config.Config) *OrganizationEventHandler {
+	return &OrganizationEventHandler{
+		repositories:         repositories,
+		log:                  log,
+		notificationProvider: NewNotificationProvider(log, cfg.Services.Novu.ApiKey),
+	}
+}
+
+type eventMetadata struct {
+	UserId string `json:"user-id"`
+}
+
+func (h *OrganizationEventHandler) OnOrganizationUpdateOwner(ctx context.Context, evt eventstore.Event) error {
+	span, ctx := opentracing.StartSpanFromContext(ctx, "Notifications.OrganizationEventHandler.OnOrganizationUpdateOwner")
+	defer span.Finish()
+	setEventSpanTagsAndLogFields(span, evt)
+
+	var eventData events.OrganizationUpdateOwnerEvent
+	if err := evt.GetJsonData(&eventData); err != nil {
+		tracing.TraceErr(span, err)
+		return errors.Wrap(err, "evt.GetJsonData")
+	}
+
+	userId := eventData.UserId
+
+	err := h.notificationProviderSendEmail(ctx, span, EventIdUserUpdate, userId, eventData.Tenant)
+
+	if err != nil {
+		tracing.TraceErr(span, err)
+	}
+
+	return err
+}
+
+func (h *OrganizationEventHandler) notificationProviderSendEmail(ctx context.Context, span opentracing.Span, eventId, userId, tenant string) error {
+
+	userDbNode, err := h.repositories.UserRepository.GetUser(ctx, tenant, userId)
+
+	if err != nil {
+		tracing.TraceErr(span, err)
+		return errors.Wrap(err, "h.repositories.UserRepository.GetUser")
+	}
+	var user *entity.UserEntity
+	if userDbNode != nil {
+		user = graph_db.MapDbNodeToUserEntity(*userDbNode)
+	}
+
+	emailDbNode, err := h.repositories.EmailRepository.GetEmailForUser(ctx, tenant, userId)
+
+	var email *entity.EmailEntity
+	if emailDbNode != nil {
```

Comment:
we should fail here. you can't sent a notification on email without an email

Reply:
indeed! hahaha missed this entirely!